### PR TITLE
SALTO-5334, SALTO-5346 - Salesforce: move PermissionSet refs to custom references code

### PR DESCRIPTION
**Pull request by tamtamirr**
This implementation is based on old PR by @tomsellek  https://github.com/salto-io/salto/pull/5547

Move the extraction of references from the reference mapper to custom references. This will speed fetches in cases where `PermissionSet`s are included and some of them are very large.

---

_Release Notes_: 
Salesforce Adapter:
- Add PermissionSet custom references handler.

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/salesforce-adapter/src/custom_references/permission_sets.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * limitations under the License.
+ */
+
+import { ReferenceInfo, Element, InstanceElement } from '@salto-io/adapter-api'
+import { logger } from '@salto-io/logging'
+import { collections } from '@salto-io/lowerdash'
+import { PERMISSION_SET_METADATA_TYPE } from '../constants'
+import { isInstanceOfTypeSync } from '../filters/utils'
+import { WeakReferencesHandler } from '../types'
+import { mapProfileOrPermissionSetSections } from './profiles'
```
Comment (yelly) [inline packages/salesforce-adapter/src/custom_references/permission_sets.ts:23]: Maybe move all the common code to a separate file?

File: packages/salesforce-adapter/src/custom_references/permission_sets.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    'Generated %d references for %d elements.',
+    refs.length,
+    elements.length,
+  )
+  return refs
+}
+
+export const permissionSetsHandler: WeakReferencesHandler = {
+  findWeakReferences,
+  removeWeakReferences: () => async () => ({ fixedElements: [], errors: [] }),
```
Comment (yelly) [inline packages/salesforce-adapter/src/custom_references/permission_sets.ts:64]: We will need to also implement fix elements here for the same reason. OK to do in a separate PR.

**Review by amitgoldarb — APPROVED** [review]
Thanks for this!

File: packages/salesforce-adapter/src/custom_references/permission_sets.ts (hunk 1 above)
Comment (tamtamirr) [reply packages/salesforce-adapter/src/custom_references/permission_sets.ts:23]: Will do in a separate PR 🙏
